lib/lltz_michelson/instruction.ml

```diff
@@ -0,0 +1,499 @@
+  Config.ok stack [ I.loop_left instrs ]
+
+(* https://tezos.gitlab.io/michelson-reference/#instr-ITER *)
+let iter (in_ : t) stack =
+  let stack =
+    match stack with
+    | `Value :: stack -> stack
+    | _ -> raise_s [%message "Instruction.iter" (stack : SlotStack.t)]
+  in
+  let instrs = Config.instructions @@ in_ (`Value :: stack) in
+  Config.ok stack [ I.iter instrs ]
+
+(* https://tezos.gitlab.io/michelson-reference/#instr-MAP *)
+let map_ in_ stack =
+  let stack =
+    match stack with
+    | `Value :: stack -> stack
+    | _ -> raise_s [%message "Instruction.map" (stack : SlotStack.t)]
+  in
+  let instrs = Config.instructions @@ in_ (`Value :: stack) in
+  Config.ok stack [ I.map instrs ]
+
+(* https://tezos.gitlab.io/michelson-reference/#instr-IF *)
+let if_ ~then_ ~else_ stack =
+  let stack =
+    match stack with
+    | `Value :: stack -> stack
+    | _ -> raise_s [%message "Instruction.if_" (stack : SlotStack.t)]
+  in
+  Config.merge (then_ stack) (else_ stack) ~f:(fun instrs1 instrs2 ->
+    [ I.if_ ~then_:instrs1 ~else_:instrs2 ])
+
+(* https://tezos.gitlab.io/michelson-reference/#instr-IF_LEFT *)
+let if_left ~left ~right stack =
+  let stack =
+    match stack with
+    | `Value :: _ -> stack
+    | _ -> raise_s [%message "Instruction.if_left" (stack : SlotStack.t)]
+  in
+  Config.merge (left stack) (right stack) ~f:(fun instrs1 instrs2 ->
+    [ I.if_left ~then_:instrs1 ~else_:instrs2 ])
+
+(* https://tezos.gitlab.io/michelson-reference/#instr-IF_CONS *)
+let if_cons ~empty ~nonempty stack =
+  let stack =
+    match stack with
+    | `Value :: stack -> stack
+    | _ -> raise_s [%message "Instruction.if_cons" (stack : SlotStack.t)]
+  in
+  Config.merge
+    (empty stack)
+    (nonempty (`Value :: `Value :: stack))
+    ~f:(fun instrs1 instrs2 -> [ I.if_cons ~then_:instrs1 ~else_:instrs2 ])
+
+(* https://tezos.gitlab.io/michelson-reference/#instr-IF_NONE *)
+let if_none ~none ~some stack =
+  let stack =
+    match stack with
+    | `Value :: stack -> stack
+    | _ -> raise_s [%message "Instruction.if_none" (stack : SlotStack.t)]
+  in
+  Config.merge
+    (none stack)
+    (some (`Value :: stack))
+    ~f:(fun instrs1 instrs2 -> [ I.if_none ~then_:instrs1 ~else_:instrs2 ])
+
+module Slot = struct
+  let def (slot : [< Slot.definable ]) ~in_ stack =
+    let stack =
+      match stack with
+      | slot' :: stack ->
+        if not (Slot.is_assignable slot' ~to_:slot)
+        then raise_s [%message "Instruction.Slot.def: slot not assignable"];
+        (slot :> Slot.t) :: stack
+      | [] ->
+        raise_s
+          [%message
+            "Instruction.Slot.bind: invalid stack"
+              (stack : SlotStack.t)
+              (slot : [< Slot.definable ])]
+    in
+    in_ stack
+
+  (* remove slot from stack, for example after it is used by let*)
+  let collect slot stack =
+    try
+      let found_slot = SlotStack.find_exn stack slot in
+      remove found_slot stack
+    with
+    | _ ->
```

Comment:
You must handle only the exception you expect! Or, better, don't use `find_exn`.

Reply:
this is actually handled inside of find_exn